packages/kit/test/apps/basics/test/test.js

```diff
@@ -323,16 +323,16 @@ test.describe('Encoded paths', () => {
 		});
 	});
 
-	test('allows %-encoded characters in directory names', async ({ page, clicknav }) => {
+	test('allows non-ASCII character in parameterized route segment', async ({ page, clicknav }) => {
 		await page.goto('/encoded');
-		await clicknav('[href="/encoded/$SVLT"]');
-		expect(await page.textContent('h1')).toBe('$SVLT');
+		await clicknav('[href="/encoded/@svelte"]');
+		expect(await page.textContent('h1')).toBe('@svelte');
 	});
 
-	test('allows %-encoded characters in filenames', async ({ page, clicknav }) => {
+	test('allows characters to be represented as HTML entities', async ({ page, clicknav }) => {
```

Comment:
some other characters worth testing as having unique properties:
- `#`
- `?`
- `*`
- `苗` (as `&#x82D7;`)
- `<`

you could probably just shove them all in this same test

Reply:
Added tests for all but `&#x82D7;` which is... tricky. We don't let you include a `#` character in a filename, because it breaks the URL as far as Vite is concerned.

So we either need to accept the limitation that only named entities can be used (which covers all the exceptions that we actually care about — the characters that can't be used in a Windows filename plus the characters that have special meaning to SvelteKit), or we need to come up with an alternative way of encoding characters. URL-encoding is a no-go, and `\u1234`-style encoding won't work because you can't have a `\` character on Windows

Reply:
we could always do something wacky like `src/routes/{u+1234}/+page.svelte`, if we reserved the `{ }` characters

Reply:
I actually quite like that, the more I think of it — maybe we use that scheme instead of HTML entities, so that there's one consistent approach? Much simpler implementation-wise.

Any other characters we should be reserving for future use?

Reply:
I'm not sure how I feel about that particular syntax versus other potential ones - but I do think it's a good move to support only ASCII/Unicode code points here, and not all possible named HTML entities.

Reply:
What other syntaxes should we consider? I think the `u+1234` bit makes sense, as that's in fairly widespread use already, so it's really just the delimiters we need to worry about. If we want symmetrical delimiters then the options are `{u+1234}`, `[u+1234]` or `(u+1234)`. I think `( )` is probably a bit weird since elsewhere it means 'ignore this'. I don't hate `[ ]` — it's not breaking, since `[u+1234]` is currently disallowed, and avoids using up another character pair we might put to some other use in future.

We could also do `|u+1234|` or whatever though that's probably a bit weird.

Reply:
`|` is disallowed in path names in Windows (and would likely cause confusion with shell escaping anyway, because of stdout/stdin piping).

`[u+1234]` not using up another delimiter just for this seems a bit compelling.